dask/dataframe/tests/test_optimize_dataframe.py

```diff
@@ -48,5 +48,19 @@ def test_fuse_ave_width():
 
     b = s.__dask_optimize__(s.dask, s.__dask_keys__())
 
-    assert len(a) < len(b)
     assert len(a) <= 15
+    assert len(b) <= 15
+
+
+@pytest.mark.xfail(reason="need better high level fusion")
```

Comment:
This is future work.  We need better fusion on the high level graph level. cc @jcrist ?

Reply:
Sure, that'd be fun to work on. Is this blocking for this PR to merge, or just future work?

Reply:
Not blocking.  Things are better with this PR (`df + 1 + 2 + 3 + 4` happens in one task) but could be much better still with better fusion.  In particular I've seen the use case in this failing test happen frequently.  It seems that common use of Pandas includes dozens of lines of modifying columns in place, all of which generate diamond-like graphs.

Reply:
The PR itself could also use review though.